### PR TITLE
Verb-like commands for "thainlp" command-line program

### LINKED ISSUE #342: CLI commands should be verbs/actions
Current pythainlp commands are `['corpus', 'tokenization', 'soundex', 'tagging']`, which are
- not verbs
- not consistent with module names (`.corpus`, `.tokenize`, `.transliterate`, `.tag`)
(see: https://github.com/PyThaiNLP/pythainlp/blob/dev/pythainlp/cli/__init__.py )

Verb commands are more intuitive (and many time also shorter). 

Example of verb commands:
- git add
- git commit
- git push
- git pull

We should try to use verb commands when possible.
Work in progress in https://github.com/PyThaiNLP/pythainlp/tree/verb-commands 

## Todo:
- `thainlp data purge` to remove all downloaded datasets
- Add `benchmark` command: https://github.com/PyThaiNLP/pythainlp/blob/dev/bin/word-tokenization-benchmark
- Write testcases (maybe using subprocess)
- Update document at https://github.com/PyThaiNLP/pythainlp/blob/verb-commands/docs/notes/command_line.rst
- Record usage demo with https://asciinema.org

### PR DESCRIPTION
**Pull request by bact**
This PR will update cli commands and close #342.

### CODE REVIEW

**Comment by wannaphong** [comment]
It's OK. Can you help to improve the command-line document?

File: pythainlp/cli/data.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+class App:
+    def __init__(self, argv):
+        parser = argparse.ArgumentParser(
+            prog="data", usage="thainlp data <subcommand>",
+        )
+        parser.add_argument(
+            "subcommand",
+            type=str,
+            # choices=["catalog", "info", "get", "rm", "path", "ls"],
```
Comment (p16i) [inline pythainlp/cli/data.py]: can we remove this line?

File: pythainlp/cli/data.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            help="action on dataset/corpus (catalog, info, get, rm, path)",
+        )
+
+        args = parser.parse_args(argv[2:3])
+        if not hasattr(App, args.subcommand):
+            print(f"Subcommand not available: {args.subcommand}")
+            parser.print_help()
+            exit(1)
```
Comment (p16i) [inline pythainlp/cli/data.py]: what do you if we raise an exception here instead?

For example, we might use `raise ValueError(...)`.

Ref: https://docs.python.org/3/library/exceptions.html#ValueError

**Comment by p16i** [comment]
How should we update the document?

**Comment by wannaphong** [comment]
> 
> 
> How should we update the document?

You can update the document in https://github.com/PyThaiNLP/pythainlp/blob/verb-commands/docs/notes/command_line.rst

File: pythainlp/cli/data.py (hunk 2 above)
Comment (bact) [reply pythainlp/cli/data.py]: Decided to go with argparser's built-in `choices=[..]`. Let them handle it.

**Comment by wannaphong** [comment]
เราสามารถใช้ subprocess ในการทำ unittest ส่วน command-line ได้ไหมครับ?

Can we use `subprocess` for test command line applications?

**Comment by p16i** [comment]
Yes, I think we can do that.

So, we can invoke our CLIs via `subprocess`, then do the computations using small datasets, and finally we check the output from the CLIs with our test cases.

Having said that, this approach is applicable only for CLIs that produce output; otherwise we have to come up with a different solution.

**Review by wannaphong — APPROVED** [review]

**Pull request merged**